.github/workflows/CI.yaml

```
@@ -24,7 +24,21 @@ jobs:
           key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
           restore-keys: |
             ${{ runner.os }}-gradle-
+      - name: Force a fresh version of neo4j (not the cached one)
+        run: (find ~/.gradle/caches -name "*neo4j*" -exec rm -rf {} \;) || echo "All neo4j files cleaned"
       - name: Grant execute permission for gradlew
         run: chmod +x gradlew
       - name: Build with Gradle
+        env:
+          TEAMCITY_DEV_URL: ${{ secrets.TEAMCITY_DEV_URL }}
+          TEAMCITY_USER: ${{ secrets.TEAMCITY_USER }}
+          TEAMCITY_PASSWORD: ${{ secrets.TEAMCITY_PASSWORD }}
         run: ./gradlew build --continue
+      - name: Clean neo4j from the cache
+        run: (find ~/.gradle/caches -name "*neo4j*" -exec rm -rf {} \;) || echo "All neo4j files cleaned"
+      - name: Archive test results
+        uses: actions/upload-artifact@v2
+        with:
+          name: test-results
+          path: /home/runner/work/apoc-private/apoc-private/core/build/reports/tests/test/index.html
```

@ncordon I think that we should be more general:
```suggestion
          path: ${{ github.workspace }}/core/build/reports/tests/test/index.html
```
Please look here: https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables

does this work also for `full` tests? I guess that they are catching only `core` now.

Maybe you can also investigate some action that prints the falling test directly into the action like this one:
https://github.com/marketplace/actions/test-reporter

---

Fair, this is not needed anymore. I've given a try to the test reporter, but I haven't managed to make it work, so we can start by having the archived test reports at least for now (I've also included `full` now) and we can iterate over this in a separate task.